lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -604,6 +604,12 @@ static bool printDirectReturnOrParamCType(
         addABIRecordToTypeEncoding(typeEncodingOS, offset, end, t, typeMapping);
       }))
     return false;
+  if (isResultType && Count == 0) {
+    // A direct result with no record members can happen for uninhabited result
+    // types like `Never`.
+    os << "void";
```

Comment:
When I was working on this PR, I noticed that IRGen represents `Never` return types as having one empty-tuple direct result, rather than as having no direct result. (This previously tripped an assert.) I'm assuming that this is correct, or at least that changing it would break the ABI, and that using a `void` return with `swiftcall` is the correct prototype to use to call such a function from C++.

I'd love to get conformation from someone who works on IRGen (maybe @aschwaighofer or @rjmccall?) that this is the right approach.